### PR TITLE
Hetzner

### PR DESCRIPTION
**Pull request by gfleury**
### Description

This is the driver for Hetzner provider (www.hetzner.de). It's a decent provider in Germany heavy used by some local startups in Germany. Basically provides bare metal and virtualized nodes. Documentation is also present on the commits. 

### Status

- done, ready for review

### Checklist (tick everything that applies)

- [X] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [X] Documentation
https://github.com/apache/libcloud/commit/ff764ebbabb1df2b7ca8af77a5542a46284f1973
- [X] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
https://github.com/apache/libcloud/commit/5724c2cfda9972e39e6f2afcfa142d5500a3137d
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

File: example_compute.py [hunk 1]
```diff
@@ -29,7 +27,7 @@
 # <Node: provider=Rackspace, status=REBOOT, name=korine, ip=6.7.8.9.10>, ... ]
 
 # grab the node named "test"
-node = [n for n in nodes if n.name == 'test'][0]
+#node = [n for n in nodes if n.name == 'test'][0]
 
 # reboot "test"
-node.reboot()
+#node.reboot()
```
Comment (pquentin) [inline example_compute.py]: Any reason for this to be commented out?

**Comment by pquentin** [comment]
Thank you, we received your ICLA. Can you please rebase your changes on top of trunk? I fixed the build issues.

File: example_compute.py (hunk 1 above)
Comment (gfleury) [reply example_compute.py]: Actually this file should not be commited, I will revert this one.

**Comment by pquentin** [comment]
Unfortunately this pull requests contains commits from the trunk branch, which makes it harder to review. Do you think you could rebase on top of trunk, and make sure you only have your commits in the pull request?

If you don't know how to do it, please give me access to your fork and I'll do it myself.

**Comment by costela** [comment]
Just FYI, this PR seems to implement the "old" API (https://robot.your-server.de/doc/webservice/en.html). There is also a newer "cloud" API (https://docs.hetzner.cloud/) that allows a more programmatic approach to managing servers (AFAICT it's not possible to create a server using the old API). However, the newer API seems to be limited to the newer cloud products, i.e.: it doesn't seem to be able to manage the bare-metal servers.

**Comment by pquentin** [comment]
@gfleury why did you close this? Is this because of the comment above?

**Comment by costela** [comment]
To clarify: I didn't mean to imply this PR should be dropped. As I mentioned, the old API is still needed to manage some products and I couldn't find any clear information about a deprecation timeline, so I think we could assume it will be maintained for the foreseeable future.

**Comment by blokhin** [comment]
Dear George @gfleury,
Dear Quentin @pquentin,

is there any chance Hetzner will be supported by Libcloud?

**Pull request closed without merging**